torchvision/models/detection/transform.py

```diff
@@ -88,21 +92,42 @@ def resize(self, image, target):
         return image, target
 
     def batch_images(self, images, size_divisible=32):
-        # concatenate
-        max_size = tuple(max(s) for s in zip(*[img.shape for img in images]))
-
-        stride = size_divisible
-        max_size = list(max_size)
-        max_size[1] = int(math.ceil(float(max_size[1]) / stride) * stride)
-        max_size[2] = int(math.ceil(float(max_size[2]) / stride) * stride)
-        max_size = tuple(max_size)
-
-        batch_shape = (len(images),) + max_size
-        batched_imgs = images[0].new(*batch_shape).zero_()
-        for img, pad_img in zip(images, batched_imgs):
-            pad_img[: img.shape[0], : img.shape[1], : img.shape[2]].copy_(img)
-
-        return batched_imgs
+        if torch._C._get_tracing_state():
```

Comment:
This function has been effectively entirely rewritten to make it work for ONNX. In this case, I think it would make sense to write a new method, `_batch_images_for_onnx`, which is dispatched instead if we are in tracing mode.

thoughts?

Reply:
Definitely.
I will include it in the next commit.